src/basic/fs-util.c

```diff
@@ -495,6 +496,34 @@ int get_files_in_directory(const char *path, char ***list) {
         return n;
 }
 
+int var_tmp(char **ret) {
+        const char *tmp_dir = NULL;
+        const char *env_tmp_dir = NULL;
+        char *c = NULL;
+        int r;
+
+        assert(ret);
+
+        env_tmp_dir = getenv("TMPDIR");
+        if (env_tmp_dir != NULL) {
+                r = is_dir(env_tmp_dir, true);
+                if (r < 0 && r != -ENOENT)
+                        return r;
+                if (r > 0)
+                        tmp_dir = env_tmp_dir;
+        }
+
+        if (!tmp_dir)
+                tmp_dir = "/var/tmp";
+
+        c = strdup(tmp_dir);
+        if (!c)
+                return -ENOMEM;
+        *ret = c;
+
+        return 0;
+}
```

Comment:
Not that it matters too much, but given that in all cases the returned string is available in static memory anyway, this function coulde actually avoid the memory allocation and just return the constant string. i.e.:

```c
int var_tmp(const char **ret);
```